### ISSUE TITLE
great job！have some questions to ask

### ISSUE THREAD

**Issue by positive666** [agentica-project/deepscaler#7]
1.Does this model currently have issues with language consistency alignment? I occasionally encounter situations where I ask questions in Chinese, but the responses are in English. I wonder if there has been any reward learning in this regard.

2. asked some questions like "surrrrrr" with multiple "r"s, and the answers sometimes are correct while other times they are not. Is this a common issue with smaller models currently?
3. 
3.The model's output is satisfactory, but I'm not sure if the response format is entirely consistent. This should be strongly related to the training data, right? I'd like to inquire about the approximate GPU memory required. For example, if I have six A100 GPUs (40G each), what would be the considerations?

**Comment by michaelzhiluo** [comment]
1. The model was only trained on 1/0 reward if the answer is correct or not. The entire dataset is english, that's probably why.
2. Bit confused wym here... Smaller models tend to have higher variance in their outputs.
3. Our training dataset for RL is only 40K points, the pretraining phase is from Qwen team, no clue what they did there. Also the model is small can fit on < 6 GB of ram for serving if needed.

**Comment by positive666** [comment]
Thank you for your response. I am testing by deploying it on my own device. However, what I would like to consult about is the resource situation for training reproducibility. After all, it is just a 1.5B model. Is the GPU memory requirement due to the increased sequence length? I am not sure if my device can handle the reproducible training. 
Some open-source solutions have explicitly stated that models smaller than 3B will have relatively poor inference capabilities. I wonder if this is because the model is based on SFT (Supervised Fine-Tuning) with Chain of Thought (CoT), and reinforcement learning (RL) is applied to address this issue through extended sequence length?

**Comment by michaelzhiluo** [comment]
I think 8k and 16K context should be fine, you can adjust the vLLM memory utilization in Verl's config!

And second, yes RL is way better for generalization in the reasoning domain than SFT, the model adapts towards its given max context length ;) Someone should write a paper on this!

**Comment by positive666** [comment]
thanks! have a few points I'd like to consult on further:
1.In my interactions with some scholars, there are two viewpoints. One is that it is good for the model to output more and more during the Chain of Thought (CoT) process. The other is that if the CoT is too long, it is not good.
2. In DeepSeek's R1, two points were mentioned. First, I don't know where their reward model in the final stage comes from. Second, how do they make the reasoning chains shorter for simple questions and longer for complex questions? Is it through rule-based rewards?

**Comment by biandh** [comment]
>
> And second, yes RL is way better for generalization in the reasoning domain than SFT, the model adapts towards its given max context length ;) Someone should write a paper on this!

I am also very interested in this because the DeepSeek-R1-Zero-Qwen-32B effect of R1's paper on 32B is not as good as the direct DeepSeek-R1-Distill-Qwen-32B effect, and your conclusion seems to contradict that. Do you have any interesting insights on this?

**Comment by positive666** [comment]
> thanks! have a few points I'd like to consult on further: 1.In my interactions with some scholars, there

> > And second, yes RL is way better for generalization in the reasoning domain than SFT, the model adapts towards its given max context length ;) Someone should write a paper on this!
> 
> I am also very interested in this because the DeepSeek-R1-Zero-Qwen-32B effect of R1's paper on 32B is not as good as the direct DeepSeek-R1-Distill-Qwen-32B effect, and your conclusion seems to contradict that. Do you have any interesting insights on this?

"I get that you're saying high-quality CoT data should go through SFT before RL or zero-shot learning. Personally, I think it's more about SFT plus RL, not just RL alone. The authors of this project built it on DeepSeek-R1-Distill-Qwen-1.5B, so it's safe to assume it's already been through SFT.”

**Comment by michaelzhiluo** [comment]
> > And second, yes RL is way better for generalization in the reasoning domain than SFT, the model adapts towards its given max context length ;) Someone should write a paper on this!
> 
> I am also very interested in this because the DeepSeek-R1-Zero-Qwen-32B effect of R1's paper on 32B is not as good as the direct DeepSeek-R1-Distill-Qwen-32B effect, and your conclusion seems to contradict that. Do you have any interesting insights on this?

We did try RL on top of Qwen-1.5B base for 4K context versus Distill-Deepseek at 4k context. It does get to similar performance but requires 3-5x more steps. Distill is like a good starting point, as Qwen base seems to already have some reasoning in its pre training ;) I'd argue that RL is all you need in this case.

**Comment by biandh** [comment]
> > > And second, yes RL is way better for generalization in the reasoning domain than SFT, the model adapts towards its given max context length ;) Someone should write a paper on this!
> > 
> > 
> > I am also very interested in this because the DeepSeek-R1-Zero-Qwen-32B effect of R1's paper on 32B is not as good as the direct DeepSeek-R1-Distill-Qwen-32B effect, and your conclusion seems to contradict that. Do you have any interesting insights on this?
> 
> We did try RL on top of Qwen-1.5B base for 4K context versus Distill-Deepseek at 4k context. It does get to similar performance but requires 3-5x more steps. Distill is like a good starting point, as Qwen base seems to already have some reasoning in its pre training ;) I'd argue that RL is all you need in this case.

Wow, Based on your experimental conclusions, the reason why the performance of R1-Zero in the R1 paper is not as good as that of R1-Distill is actually just due to insufficient iterations, In other words, if the number of RL iterations steps is sufficient, R1-Zero's performance would be better than that of R1-Distill. Is that so?

**Comment by michaelzhiluo** [comment]
Not necessarily, they trained even longer context (im guessing up to 64-128K).  With such long contexts, comes more challenges due to extreme sparsity of the chain of through optimization space.
